### ISSUE TITLE
ListMap: Does not implement operator==

### ISSUE THREAD

**Issue by vHanda** [marcglasberg/fast_immutable_collections#35]
I understand that it's mutable, unlike the others. However, considering that all other collections implement a proper `operator ==`, I expected ListMap to do the same.

Maybe we can make the documentation reflect that?

```dart
test('ListMap Equality', () {
    var map1 = ListMap.of({
      'b': 1,
      'a': 2,
    });

    var map2 = ListMap.of({
      'b': 1,
      'a': 2,
    });

    expect(map1, map2);
    expect(map1 == map2, true);
  });
```

The second expect fails.

**Comment by marcglasberg** [comment]
Yes, `ListMap` is mutable, so I followed Dart's convention for collections, which is that equality is by **identity**.

Implementing equality by checking each item in the collection can be very slow for large collections, while using identity is instantaneous. In `IList`, `ISet` and `IMap` I actually offer both possibilities (by choosing `ConfigList`, `ConfigSet` and `ConfigMap`), because both are useful depending on the situation.

The `ListMap` was actually created because I needed it to implement the `IMap`, but I opened it for others to use because it can be useful by itself. So I don't intend on further evolving it.

However, it's very easy for you to create your own `MyListMap` which does what you want. Just extend `ListMap` and override `operator ==` and `hashCode()`:

Use  `const MapEquality().equals(list, other) for the `operator ==`.

Use `const MapEquality().hash(list) for the `hashCode()`.

Then, run our `expect(map1 == map2, true);` to see that it works.
